torch_xla/csrc/tensor.cpp

```diff
@@ -120,14 +120,15 @@ XLATensor::XLATensor(std::shared_ptr<View> view,
                                        logical_element_type)) {}
 
 XLATensor::XLATensor(std::shared_ptr<Data> data)
-    : data_(std::move(data)),
+    : torch::lazy::LazyTensor(data),
+      data_(std::move(data)),
       storage_(c10::Storage(
           {}, 0,
           c10::DataPtr(nullptr, backendDeviceToAtenDevice(data_->device)))) {}
 
-XLATensor::Data* XLATensor::data() const {
-  XLA_CHECK(data_ != nullptr) << "Trying to access a null cursor";
-  return data_.get();
+auto XLATensor::data() const -> const std::shared_ptr<Data>& {
+  TORCH_CHECK(data_ != nullptr, "Trying to access a null cursor");
```

Comment:
I think we should use `XLA_CHECK` within our repo. Unless we want to replace all `XLA_CHECK` with `TORCH_CHECK`

Reply:
Good catch. Let me fix that.